triage_bot/lib/engine.dart

```diff
@@ -0,0 +1,1499 @@
+                  await github.issues.addLabelsToIssue(GitHubSettings.primaryRepository, number, <String>[GitHubSettings.staleIssueLabel]);
+                  await _githubReady();
+                  await github.issues.createComment(GitHubSettings.primaryRepository, number,
+                    'This issue is assigned to $assignee but has had no recent status updates. '
+                    'Please consider unassigning this issue if it is not going to be addressed in the near future. '
+                    'This allows people to have a clearer picture of what work is actually planned. Thanks!',
+                  );
+                } else if (issue.lastContributorTouch!.isBefore(reallyStaleThreshold)) {
+                  bool skip = false;
+                  String team = 'primary triage';
+                  if (assignedTeams.length == 1) { // if it's more, then cleanup will take care of it
+                    team = assignedTeams.single;
+                    if (!_lastRefeedByTime.containsKey(team) || _lastRefeedByTime[team]!.isBefore(refeedThreshold)) {
+                      _lastRefeedByTime[team] = now;
+                    } else {
+                      skip = true;
+                    }
+                  }
+                  if (!skip) {
+                    log('Issue #$number is assigned to $assignee but still not making progress (for ${now.difference(issue.lastContributorTouch!)}); sending back to triage (for $team team).');
+                    for (final String triagedTeam in triagedTeams) {
+                      await _githubReady();
+                      await github.issues.removeLabelForIssue(GitHubSettings.primaryRepository, number, '${GitHubSettings.triagedPrefix}$triagedTeam');
+                    }
+                    await _githubReady();
+                    await github.issues.edit(GitHubSettings.primaryRepository, number, IssueRequest(assignees: const <String>[]));
+                    await _githubReady();
+                    await github.issues.createComment(GitHubSettings.primaryRepository, number,
+                      'This issue was assigned to $assignee but has had no status updates in a long time. '
+                      'To remove any ambiguity about whether the issue is being worked on, the assignee was removed.',
+                    );
+                    await _githubReady();
+                    await github.issues.removeLabelForIssue(GitHubSettings.primaryRepository, number, GitHubSettings.staleIssueLabel);
+                  }
+                }
+              }
+            }
+            // Check for P1 issues that aren't making progress.
+            // We currently rate-limit this to only a few per week so that teams don't get overwhelmed.
+            if (issue.assignedAt == null &&
+                issue.labels.contains('P1') &&
+                issue.lastContributorTouch != null &&
+                issue.lastContributorTouch!.isBefore(staleThreshold) &&
```

Comment:
Considering we want bug to be updated on a monthly basis for P1, shall we use a tighter threshold? Now we seem to use 20*7 days for all stale cases.

Reply:
I expect we'll want to tighten this up over time, but I'm hoping to start very conservative here.